extension/json/src/json_scan.cpp

```
@@ -74,8 +74,19 @@ struct JsonBindData : public ScanBindData {
             scanFromList, scanFromStruct, config.copy(), context);
     }
 
-    uint32_t getIdxFromName(const std::string& s) const {
-        return nameToIdxMap.at(StringUtils::getUpper(s));
+    int32_t getIdxFromName(std::string s) const { // possible bottleneck
+        auto cpy = StringUtils::getUpper(s);
+        if (!nameToIdxMap.contains(cpy)) {
+            // try removing any [a-zA-Z]+\. prefix
+            auto periodPos = cpy.find('.');
+            if (periodPos != std::string::npos) {
+                cpy = cpy.substr(periodPos + 1);
+            }
+        }
+        if (!nameToIdxMap.contains(cpy)) {
+            return -1;
```

Should we throw an exception if we don't find the field name